syft/pkg/cataloger/common/generic_cataloger.go

```diff
@@ -31,6 +33,16 @@ func NewGenericCataloger(pathParsers map[string]ParserFn, globParsers map[string
 	}
 }
 
+// NewPostProcessingGenericCataloger if provided path-to-parser-function and glob-to-parser-function lookups creates a GenericCataloger
+func NewPostProcessingGenericCataloger(pathParsers map[string]ParserFn, globParsers map[string]ParserFn, upstreamCataloger string, postProcess PostProcessFunc) *GenericCataloger {
```

Comment:
open question: we could allow for one or more of these processors and make it variadic. This would also allow for a common constructor:
```
func NewGenericCataloger(pathParsers map[string]ParserFn, globParsers map[string]ParserFn, upstreamCataloger string, processors ...PostProcessFunc) *GenericCataloger {
...
}
```